packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -1,202 +1,155 @@
+  access_token: z.string(),
+  expiration_date: z.string(),
+});
 
-type EarthdataPostTokenResponse = Response<{
-  body: {
-    access_token?: string,
-    expiration_date?: string
-  }
-}>;
+const PostTokenSchema = z.object({
+  access_token: z.string(),
+  token_type: z.string(),
+  expiration_date: z.string(),
+});
 
-export interface EarthdataLoginParams {
-  username: string,
-  password: string,
-  edlEnv: string,
-}
+type Token = z.infer<typeof TokenSchema>;
 
-/**
- * A class to simplify requests for the Earthdata Login token used for CMR
- *
- * @typicalname earthdataLogin
- *
- * @example
- * const { EarthdataLogin } = require('@cumulus/cmr-client');
- *
- * const earthdataLogin = new EarthdataLogin({
- *  "username": "my-username",
- *  "password": "my-password",
- *  "edlEnv": "my-edl-environment"
- * });
- */
-export class EarthdataLogin {
-  username: string;
-  password: string;
-  edlEnv: string;
+const GetTokenResponseBody = z.array(TokenSchema);
 
-  /**
-  * The constructor for the EarthdataLogin class
-  *
-  * @param {string} params.username - Earthdata Login username, required parameter
-  * @param {string} params.password - Earthdata Login password, required parameter
-  * @param {string} params.edlEnv - the Earthdata Login environment (['PROD', 'OPS', 'SIT', 'UAT'])
-  *
-  * @example
-  *
-  * {
-  *  "username": "janedoe",
-  *  "password": "password",
-  *  "edlEnv": "UAT"
-  * }
-  */
+const PostTokenResponseBody = z.array(PostTokenSchema);
 
-  constructor(params: EarthdataLoginParams) {
-    this.username = params.username;
-    this.password = params.password;
-    this.edlEnv = params.edlEnv;
-  }
-
-  /**
-   * The method for getting the Earthdata Login endpoint URL based on the EDL environment
-   *
-   * @returns {Promise.<string>} the endpoint URL
-   */
-  getEDLurl(
-  ) {
-    switch (this.edlEnv) {
-      case 'PROD':
-      case 'OPS':
-        return 'https://urs.earthdata.nasa.gov';
-      case 'UAT':
-        return 'https://uat.urs.earthdata.nasa.gov';
-      case 'SIT':
-      default:
-        return 'https://sit.urs.earthdata.nasa.gov';
-    }
-  }
+/** This method is for parsing a caught error which is not an HTTPerror
+ * in case the EDL endpoint call results in an unexpected error
+ *
+ * @returns {Error}
+ */
+const parseCaughtError = (e: unknown): Error => (e instanceof Error ? e : new Error(`${e}`));
 
-  /**
-   * The method that determines if a user has a token, to call the retrieveEDLToken function
-   * which makes an API call to the Earthdata Login endpoint, or to create the token with the
-   * createEDLToken function. Returns the token as a string.
-   *
-   * @returns {Promise.<string>} the token
-   */
-  async getEDLToken(): Promise<string> {
-    let token = await this.retrieveEDLToken();
-    if (token === undefined) {
-      token = await this.createEDLToken();
-    }
-    return token;
+/**
+* The method for getting the Earthdata Login endpoint URL based on the EDL environment
```

Comment:
The indentation of these JSDocs needs to be fixed. The `*` here should be under the first `*` above.